pkg/workloadapi/workload.proto

```diff
@@ -215,6 +216,33 @@ enum TunnelProtocol {
   // Future options may include things like QUIC/HTTP3, etc.
 }
 
+// NativeTunnel describes how an application or gateway
+// can consume tunnel information directly.
+message NativeTunnel {
+  // A target natively handles this type of traffic.  
+  GatewayProtocol protocol;
+  // optional: if set, traffic should be sent to this port after the last zTunnel hop
+  uint32 port;
+}
+
+enum GatewayProtocol {
+  // Bytes are just copied directly.
+  NONE = 1;
+
+  // The workload nativley supports HBONE
+  // For the zTunnel, this means:
+  // * Requests *from* this workload do not need to be tunneled if they already are tunneled by the tunnel_protocol.
+  // * Requests *to* this workload, via the tunnel_protocol, do not need to be de-tunneled.
+  HBONE = 2;
```

Comment:
We plausible would want a mode where we terminate TLS-HBONE, and re-initiate a new tunnel with CONNECT, probably over plaintext. Basically like PROXY but with CONNECT. Maybe we just call that CONNECT later though if we add that.

Aside -- why did we pick PROXY over this in the first place? Both http1 connect and proxy are basically just a static prefix on the connection... though I think in envoy its probably going through the full HTTP machinery so may be quite a bit more complex/less efficient

Reply:
Can you elaborate on that scenario; who is sending TLS-HBONE, where are we terminating it and who are we authenticating as in the new tunnel?

Reply:
I think PROXY ends up being simpler in Envoy.

Reply:
I mean: `client ztunnel --> HBONE --> waypoint ztunnel, terminate --CONNECT--> waypoint envoy`.

Instead of `client ztunnel --> HBONE --> waypoint ztunnel, terminate --PROXY--> waypoint envoy`.

I don't think we need to do it, was just curious the motivation behind using PROXY instead and wanted to make sure we keep our options open (we do; just a new enum value)